### PR TITLE
Document UseHttps limitations for Http/3 in Kestrel

### PR DESCRIPTION
**Pull request by wtgodbe**
Fixes https://github.com/dotnet/aspnetcore/issues/26415

### CODE REVIEW

File: aspnetcore/fundamentals/servers/kestrel/http3.md [hunk 1]
```diff
@@ -74,3 +74,14 @@ For more information on building the host, see the **Set up a host** and **Defau
...
+
+## Limitations
+
+Certain HTTPS scenarios are not yet supported for HTTP/3 in Kestrel. When calling <xref:Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps> with <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> while using HTTP/3, setting the following options on the <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> will no-op:
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions.HandshakeTimeout>
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions.OnAuthenticate>
+
+Calling the following implementations of <xref:Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps> will throw an error when using HTTP/3:
+
+`UseHttps(this ListenOptions listenOptions, ServerOptionsSelectionCallback serverOptionsSelectionCallback, object state, TimeSpan handshakeTimeout)`
```
Comment (wtgodbe) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:86]: @Rick-Anderson can I xref to specific implementations of a method?

**Review by Tratcher — APPROVED** [review]

File: aspnetcore/fundamentals/servers/kestrel/http3.md (hunk 1 above)
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/servers/kestrel/http3.md:86]: Yes, but you usually use the [Docs Authoring Pack](https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack)

Feel free to update an issue or create a new issue and assign to me.

File: aspnetcore/fundamentals/servers/kestrel/http3.md [hunk 2]
```diff
@@ -74,3 +74,14 @@ For more information on building the host, see the **Set up a host** and **Defau
 * `HttpClient` can be used for localhost/loopback testing in .NET 6 or later. Extra configuration is required when using `HttpClient` to make an HTTP/3 request:
   * Set `HttpRequestMessage.Version` to 3.0, or
   * Set `HttpRequestMessage.VersionPolicy` to `HttpVersionPolicy.RequestVersionOrHigher`.
+
+## Limitations
+
+Certain HTTPS scenarios are not yet supported for HTTP/3 in Kestrel. When calling <xref:Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps> with <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> while using HTTP/3, setting the following options on the <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> will no-op:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:80]:
```suggestion
Some HTTPS scenarios are not yet supported for HTTP/3 in Kestrel. When calling `Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps` with <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> while using HTTP/3, setting the following options on the <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> is a no-op (it does nothing):
```

File: aspnetcore/fundamentals/servers/kestrel/http3.md [hunk 3]
```diff
@@ -74,3 +74,14 @@ For more information on building the host, see the **Set up a host** and **Defau
...
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions.HandshakeTimeout>
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions.OnAuthenticate>
+
+Calling the following implementations of <xref:Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps> will throw an error when using HTTP/3:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:84]:
```suggestion
Calling the following implementations of `Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps` throw an error when using HTTP/3:
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/fundamentals/servers/kestrel/http3.md (hunk 1 above)
Comment (wtgodbe) [reply aspnetcore/fundamentals/servers/kestrel/http3.md:86]: https://github.com/dotnet/AspNetCore.Docs/issues/23432

**Comment by wtgodbe** [comment]
@Rick-Anderson any other concerns?

**Comment by JamesNK** [comment]
This section needs work. The formatting is bad:

image

**Pull request merged**